controllers/system_controller_suite_test.go

```diff
@@ -312,14 +256,218 @@ var _ = Describe("System controller", func() {
+						Tag:  pointer.StringPtr("newTag"),
+					}
+					system.Spec.Config.Rails = &saasv1alpha1.SystemRailsSpec{
+						Console: pointer.Bool(true),
+					}
+					return k8sClient.Patch(context.Background(), system, patch)
 				}, timeout, poll).ShouldNot(HaveOccurred())
-			}
+			})
+
+			It("creates the required console resources", func() {
+
+				sts := &appsv1.StatefulSet{}
+				By("deploying the system-console StatefulSet",
+					checkResource(sts, expectedResource{
+						Name: "system-console", Namespace: namespace,
+					}),
+				)
+				Expect(sts.Spec.Template.Spec.Containers[0].Image).Should((Equal("newImage:newTag")))
+				Expect(sts.Spec.Template.Spec.Volumes[0].Secret.SecretName).Should((Equal("system-config")))
+
+				pdb := &policyv1beta1.PodDisruptionBudget{}
+				By("ensuring the system-console PDB",
+					checkResource(pdb, expectedResource{
+						Name: "system-console", Namespace: namespace, Missing: true,
+					}),
+				)
+
+				hpa := &autoscalingv2beta2.HorizontalPodAutoscaler{}
+				By("ensuring the system-console HPA",
+					checkResource(hpa, expectedResource{
+						Name: "system-console", Namespace: namespace, Missing: true,
+					}),
+				)
+
+			})
+
+		})
+
+		When("updating a System resource with canary", func() {
+
+			BeforeEach(func() {
+				Eventually(func() error {
+					err := k8sClient.Get(
+						context.Background(),
+						types.NamespacedName{Name: "instance", Namespace: namespace},
+						system,
+					)
+					Expect(err).ToNot(HaveOccurred())
+					patch := client.MergeFrom(system.DeepCopy())
+					system.Spec.App = &saasv1alpha1.SystemAppSpec{
+						Canary: &saasv1alpha1.Canary{
+							ImageName: pointer.StringPtr("newImage"),
+							ImageTag:  pointer.StringPtr("newTag"),
+							Replicas:  pointer.Int32Ptr(2)},
+					}
+					system.Spec.SidekiqDefault = &saasv1alpha1.SystemSidekiqSpec{
+						Canary: &saasv1alpha1.Canary{
+							ImageName: pointer.StringPtr("newImage"),
+							ImageTag:  pointer.StringPtr("newTag"),
+							Replicas:  pointer.Int32Ptr(2)},
+					}
+					system.Spec.SidekiqBilling = &saasv1alpha1.SystemSidekiqSpec{
+						Canary: &saasv1alpha1.Canary{
+							ImageName: pointer.StringPtr("newImage"),
+							ImageTag:  pointer.StringPtr("newTag"),
+							Replicas:  pointer.Int32Ptr(2)},
+					}
+					system.Spec.SidekiqLow = &saasv1alpha1.SystemSidekiqSpec{
+						Canary: &saasv1alpha1.Canary{
+							ImageName: pointer.StringPtr("newImage"),
+							ImageTag:  pointer.StringPtr("newTag"),
+							Replicas:  pointer.Int32Ptr(2)},
+					}
+					return k8sClient.Patch(context.Background(), system, patch)
+				}, timeout, poll).ShouldNot(HaveOccurred())
+			})
+
+			It("creates the required canary resources", func() {
+
+				dep := &appsv1.Deployment{}
+				By("deploying a system-app-canary workload",
+					checkWorkloadResources(dep,
+						expectedWorkload{
+							Name:          "system-app-canary",
+							Namespace:     namespace,
+							Replicas:      1,
```

Comment:
The previous patch is setting `replicas=2`. Why test is passing with `replicas=1 `(applies to all canary tests)?

Reply:
fixed in https://github.com/3scale-ops/saas-operator/pull/142/commits/e9baecb3d539b4112dd2e777d109574d201b9aee